### ISSUE TITLE
Paper only in PDF?

### ISSUE THREAD

**Issue by iherman** [ReScience/ReScience#16]
I have found the (only) publication of the journal: [Interaction between cognitive and motor…](https://github.com/ReScience-Archives/ReScience-Entry-Topalidou-Rougier-2015/raw/1.0/article/article.pdf) but, to my great disappointment, the paper itself, ie, the narrative content, is only available in PDF. Based on the URL of the PDF file, I also got to the [article's repositiory](https://github.com/ReScience-Archives/ReScience-Entry-Topalidou-Rougier-2015), but that repository only contained a PDF as a finished text. I would have expected to find, through github.io, a version in HTML, properly readable on mobile environment, possibly directly linking into, say iPython notebooks or whatever else, etc. This journal has the potential of showing all the power of the Web for publishing...

**Comment by rougier** [comment]
You're right but it is not our top priority for the time being (lack of human resources).
However, this initiative that @ThomasA  pointed to might be interesting: https://github.com/PeerJ/paper-now

**Comment by iherman** [comment]
> On 07 Sep 2015, at 17:14 , Nicolas P. Rougier <notifications@github.com> wrote:
> 
> You're right but it is not our top priority for the time being (lack of human resources).
> However, this initiative that @ThomasA <https://github.com/ThomasA> pointed to might be interesting: https://github.com/PeerJ/paper-now <https://github.com/PeerJ/paper-now>
> 

Indeed (I didn't know about that one either).

And I of course understand the resource issues… As long as you keep this issue open, I am of course perfectly fine. Eventually...

Thanks

**Comment by antoine-lizee** [comment]
I tried to dig into this a little, and with no surprise, there is no quick win. The goal is to transform the *.md into a html fragment that could be then directly be included in the website.
I didn't find any robust, non-buggy avenue to go from the .tex file to some .html.
It seems that the answer will come from a standard .md -> .html transformation with some plugins (for cross references & bibliography) and a nice .css file. PeerJ/paper-now could be a good inspiration, but there is some work ahead anyway.

I'm +1'ing this though. It seems quite important to me.

**Comment by rougier** [comment]
I had some experiment using rst and docutils, see:

http://www.labri.fr/perso/nrougier/coding/article/article.html
http://www.labri.fr/perso/nrougier/coding/article/article.rst.html

but it is only for restructured text format and does not allow to produce the pdf. 

What about pandoc itself ? Is the produced html not good enough ?

**Comment by antoine-lizee** [comment]
Well it doesn't include any styling, so you end up with basic markup. It doesn't support the cross references nor the equations. I'm sure there is a lot of solutions, but pandoc out of the box is not going to cut it.

**Comment by rougier** [comment]
Switching to rmarkdown would require authors to have it installed and I don't know to what extent people are ready to do that (I know we're already imposing them the pandoc framework) ? Do you have any experience with rmarkdown ? Does it require R studio or can we have a lighter installation ?

**Comment by antoine-lizee** [comment]
I don't think it's really heavier than installing Haskell & cabal which is the current proposition. You do not need RStudio to compile the document, just R and the package `rmarkdown`. I just typed the following in my R console:
```
rmarkdown::render("your_article_name.md")
```
And it gave me a [decent HTML](https://htmlpreview.github.io/?https://github.com/antoine-lizee/ReScience-submission/blob/toHtml/article/your_article_name.html) from scratch. It'd be needed to create our own template, but it seems slightly easier to start from there.

**Comment by eddelbuettel** [comment]
> just R and the package rmarkdown.

It still needs a pandoc binary.

**Comment by eddelbuettel** [comment]
The one real advantage of rmarkdown as a format is that thanks to pandoc you get any one of

      html   or    pdf (via latex, of course)    or    doc (which we can IMHO ignore)

In practice one tends to pick either pdf or html early and add specialisation.   Lastly, I am firmly behind the use of (neatly typeset) pdf.  To me, html rendering depends way to much on the browser used.

**Comment by antoine-lizee** [comment]
yes, which is conveniently shipped with the package if I remember well - in any case, lighter than installing Haskell (several gigs and minutes of install). But again, that might not be the right solution - any other idea?

**Comment by antoine-lizee** [comment]
Yes, exactly: it provides a nice all-in-one, including rendering of equations, and I remember having the cross referencing working at some point too.

**Comment by eddelbuettel** [comment]
@antoine-lizee: I am sorry, but you are still confused.

RStudio (either the server or desktop) happens to install a statically built pandoc binary.  So if you have RStudio, you are good.

However, RStudio as authors of CRAN packages are _way_ to cognizant of these dependencies and the rmarkdown package does _not_ depend on RStudio. It simply states that you need a pandoc binary in the path.  Which can be a pain for some systems.  In practice, Debian/Ubuntu, brew, ... all make it possible.

**Comment by antoine-lizee** [comment]
Yep - I didn't catch that the dependency was installed by RStudio. But my point is even downloading a [pandoc binary](https://github.com/jgm/pandoc/releases/tag/1.15.0.6) is way easier than installing Haskell, which is needed for the pandoc-crossref filter only. Sorry if I haven't been clear.

**Comment by eddelbuettel** [comment]
Yup, no worries.  Now, for the matter at hand, and while I am one of R fanboys here, I think the existing workflow is good.  rmarkdown is a little too far out there for non-R users methinks...

**Comment by awakenting** [comment]
Hello,
in the context of this issue I wanted to point you to [PythonTex](http://iopscience.iop.org/article/10.1088/1749-4699/8/1/014010/meta;jsessionid=7D3C5650A8C900549E52D42BF0999ACB.c3.iopscience.cld.iop.org) as it fits the purpose of this journal pretty well, I think. To shortly summarize, it's implemented as a Latex package and allows you to include python code (ruby, octave and julia are also supported, R unfortunately not (yet)) and also execute the python code and access the output of the code. The linked article motivates the features and obviously gives a much better overview.
I realize that this not top priority and the human resources issue probably also did not magically disappear but I would be keen to hear your opinion about it.

**Comment by rougier** [comment]
Thanks for the reference. If I understand it well, it allows to embed the output of script directly into the pdf document, is that right ? Do you have any experience with it ?

However, this does not solve the HTML output "problem" but I suspect pandoc might be the solution.
Another possibility could be to use mardkeep (https://casual-effects.com/markdeep/) which seems to be quite powerful but I don't know if it exports to pdf and we might want to avoid "exotic" markup language.

**Comment by awakenting** [comment]
Yes it allows to embed the output directly. The ['Step-by-Step solution' section in [this demo pdf](https://github.com/gpoore/pythontex/blob/master/pythontex_gallery/pythontex_gallery.pdf) illustrates it quite nicely. I have no experience yet because I also just found it while searching for a way to automatically include all figures from a given directory (which is straightforward with it).

Yes, there is a depythonize utility that replaces python code by the output and gives you a plain latex document that you can convert with pandoc.

Markdeep looks interesting, indeed!

**Comment by khinsen** [comment]
I played with Markdeep a while ago. It's quite powerful, and easy to deploy as all there is to copy around is a single HTML file with embedded JavaScript and the Markdown contents. However, for that same reason Markdeep is Web-only; I see no way to get PDF or whatever else from it.

**Comment by khinsen** [comment]
After a quick look at PythonTeX, I'd say it works much like a bunch of similar tools, such as [emacs/org-mode](https://www.jstatsoft.org/article/view/v046i03), [Lepton](http://www.math.univ-paris13.fr/~lithiao/ResearchLepton/Lepton.html). A single combined input contains text and code sections, from which the code is extracted and run, with the output being integrated into the final document.

All these tools are fine for personal use, but they all suffer from the same problem in the context of publication and reviewing: they depend on a monstrous amount of dependencies and require tricky platform-dependent configuration. The name PythonTeX says it all: you must install Python and TeX, which is already a non-trivial task, and then install add-ons on both sides plus edit some configuration files to make it all work together. I wouldn't mind that effort for getting a nice tool for preparing my own publication, but as a reviewer, I expect a much lower overhead.
